htdocs/core/db/DoliDB.class.php

```diff
@@ -297,4 +297,70 @@ public function lastqueryerror()
 	{
 		return $this->lastqueryerror;
 	}
+
+
+	/**
+	 * Return first result value from query
+	 * Note : This method executes a given SQL query and retrieves the first value of the first row of results. It should only be used with SELECT queries.
+	 * Dont add LIMIT to your query, it will be added by this method
+	 * @param string $sql the sql query string
+	 * @return bool| var
+	 */
+	public function getValue($sql)
+	{
+		$sql .= ' LIMIT 1;';
+
+		$res = $this->query($sql);
+		if ($res)
+		{
+			$Tresult = $this->fetch_row($res);
```

Comment:
I fight since several years to remove completely the method fetch_row (there is still some occurences in code but not so much). The fetch_row is a method that make someting not clear. I prefer the fetch_object, where each field is explicitely named by developer instead of getting one field wihtout knowing which one when reading code.

Reply:
i agree with you but for this case it's really useful to return the first element with $Tresult[0]; it's like fetch_row was designed for that :-)

Reply:
The only option i found to not use $this->fetch_row($res); is to do that : 
```
$obj = $this->fetch_object($res);
if(is_object($obj)){
     return reset(get_object_vars($obj));
}
```

Reply:
Why not, at least it's clear that you ant the first elem. 
But what is the use case that need the first element of a SQL request when you are supposed to know which field you call by making your request.

Reply:
Finaly I have deleted this method, getRow do the same job.
Generally I use getValue for COUNT() or for link table